[Dynamic Buffer Calculation] Tolerance the scenario where the minigraph is empty

stephenxs opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Tolerance the scenario whether the minigraph is empty

Signed-off-by: Stephen Sun <stephens@nvidia.com>

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)


### Back port request
- [ ] 201911

### Approach
#### What is the motivation for this PR?

#### How did you do it?
Dynamic buffer calculation test fetches information from the minigraph.
However, in some scenarios, the testbed is not configured via using the minigraph.
In that case, we will fetch the information from CONFIG_DB

#### How did you verify/test it?
Run the regression test.

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

keboliu commented:
regarding the log level, since they are all failure cases, maybe change it to error log?

nazariig commented on tests/platform/mellanox/mellanox_psu_controller.py:
```diff
@@ -41,7 +41,7 @@ def connect_mellanox_server():
         mellanox_server.connect("a_mellanox_server", username="username", password="password")
     except Exception as e:
-        logging.debug("Failed to connect to mellanox server, exception: " + repr(e))
+        logging.warning("Failed to connect to mellanox server, exception: " + repr(e))
```
@stephenxs In terms of logging severity - warning is something not critical that system can recover from. In this case, test won't be able to recover when SSH returned error, so this should be treated as error. IMHO, the whole concept is incorrect.

stephenxs commented:
Will address the requirement in other PR/branches. Closed.

stephenxs committed:
Tolerance the scenario whether the minigraph is empty

Dynamic buffer calculation test fetch information from minigraph.
However, in some scenario the test bed is not configured via using minigraph.
In that case, we will fetch the information from CONFIG_DB

Signed-off-by: Stephen Sun <stephens@nvidia.com>
